instrumentation/dubbo/dubbo-apache-2.7/library/src/main/java/io/opentelemetry/instrumentation/dubbo/apache/v2_7/client/TracingClientFilter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.dubbo.apache.v2_7.client;
+
+import static io.opentelemetry.instrumentation.dubbo.apache.v2_7.client.DubboInjectAdapter.SETTER;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.tracer.utils.NetPeerUtils;
+import io.opentelemetry.instrumentation.dubbo.apache.v2_7.common.DubboHelper;
+import org.apache.dubbo.rpc.Filter;
+import org.apache.dubbo.rpc.Invocation;
+import org.apache.dubbo.rpc.Invoker;
+import org.apache.dubbo.rpc.Result;
+import org.apache.dubbo.rpc.RpcContext;
+import org.apache.dubbo.rpc.RpcException;
+import org.apache.dubbo.rpc.RpcInvocation;
+
+public class TracingClientFilter implements Filter {
+
+  private final DubboClientTracer tracer;
+
+  public TracingClientFilter() {
```

Comment:
```suggestion
  private TracingClientFilter() {
```

Since we have factory method

Reply:
Dubbo automatically loads filters based on their name by SPI. If this is set to private, the library will not work.Maybe I should remove the factory method?

Reply:
Ah ok - yeah let's remove the factory then.